api/v1alpha1/service.go

```diff
@@ -0,0 +1,97 @@
+package v1alpha1
+
+import (
+	"context"
+	"errors"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Service interface {
+	GetFirstLevelAccountForAccount(ctx context.Context, accountKey client.ObjectKey) (*Account, error)
+	GetFirstLevelAccountForNamespace(ctx context.Context, namespace string) (*Account, error)
+
+	GetAccount(ctx context.Context, accountKey client.ObjectKey) (*Account, error)
+	GetAccountForNamespace(ctx context.Context, namespace string) (*Account, error)
+}
+
+var _ Service = (*service)(nil)
+
+type service struct {
+	client        client.Client
+	rootNamespace string
+}
+
+func NewService(client client.Client, rootNamespace string) *service {
+	return &service{
+		client:        client,
+		rootNamespace: rootNamespace,
+	}
+}
+
+func (s *service) GetFirstLevelAccountForAccount(ctx context.Context, accountKey client.ObjectKey) (*Account, error) {
+	return s.GetAccountForNamespace(ctx, accountKey.Namespace)
+}
+
+func (s *service) GetFirstLevelAccountForNamespace(ctx context.Context, namespace string) (*Account, error) {
+	var ns corev1.Namespace
```

Comment:
should we create a helper function for this ns lookup? Seems this is repeated below

Reply:
good idea - i refactored it and put it into its own helper function